### PR TITLE
Policies for sandbox/ and scripts/ code, together with a process for moving code into scripts/

### PR DESCRIPTION
**Pull request by ctb**
A draft policy to reflect the issue raised in #471. Ready for review!

standard checklist:
- [x] Is it mergable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

And I'm also adding:

- [x] policy document accurately reflects @mr-c's wishes, perhaps after appropriate cognitive alterations have been performed

### CODE REVIEW

**Comment by ctb** [comment]
ready for review and discussion.

File: doc/dev/scripts-and-sandbox.txt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Sandbox script requirements and suggestions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in ``sandbox/`` must:
+
+* be importable (enforced by ``test_import_all`` in
+  ``test_sandbox_scripts.py``)
+* be mentioned in ``sandbox/README.rst``
+* have a shebang line ``#! /usr/bin/env python``
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]:
```
* have a hashbang line at the beginning:  ``#! /usr/bin/env python2``
```

File: doc/dev/scripts-and-sandbox.txt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+* be command-line executable (``chmod +x``)
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: `chmod u+x`

File: doc/dev/scripts-and-sandbox.txt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* have a Copyright message
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: which copyright message?

File: doc/dev/scripts-and-sandbox.txt [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+All *new* scripts being added to ``sandbox/`` should:
+
+* have decent automated tests
+* be used in a protocol (see khmer-protocols) or a recipe (see khmer-recipes)
+* be pep8 and pylint nice (see ``make pep8`` and ``make_diff_pylint``.
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: pep8 clean and pylint clean-ish

File: doc/dev/scripts-and-sandbox.txt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Command line standard options for scripts/
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in scripts/ should have the following options:
+
+* ``--force``
+* ``--loadtable`` and ``--savetable`` where appropriate (see khmer_args.py)
+
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: + ``--version``

File: doc/dev/scripts-and-sandbox.txt [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Copy or paste this checklist into the PR, in addition to the normal
+development/PR checklist::
+
+   - [ ] most or all lines of code are covered by automated tests (see output of ``make diff-cover``)
+   - [ ] ``make diff_pylint`` is clean
+   - [ ] the script has been updated with a ``get_parser()`` and added to doc/user/scripts.txt
+   - [ ] argparse help text exists, with an epilog docstring, with examples and options
+   - [ ] standard command line options are implemented
+   - [ ] version and citation information is output (`khmer_args.info`)
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: version and citation information is output to STDERR ...
and all diagnostic info to STDERR

**Comment by ctb** [comment]
jenkins please test this

**Comment by ctb** [comment]
jenkins ok to test

**Comment by ctb** [comment]
jenkins please test this

**Comment by ctb** [comment]
test this please

**Comment by ctb** [comment]
test this please, jenkins

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by ctb** [comment]
ready for review

File: doc/dev/scripts-and-sandbox.txt [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+Sandbox script requirements and suggestions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in ``sandbox/`` must:
+
+* be importable (enforced by ``test_import_all`` in
+  ``test_sandbox_scripts.py``)
+* be mentioned in ``sandbox/README.rst``
+* have a hash-bang line (``#! /usr/bin/env python``) at the top
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: python2

File: doc/dev/scripts-and-sandbox.txt [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+* be used in a protocol (see khmer-protocols) or a recipe (see khmer-recipes)
+* be pep8 clean and pylint clean-ish (see ``make pep8`` and ``make_diff_pylint``).
+
+Command line standard options for scripts/
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All scripts in scripts/ should have the following options, if they could apply:
+
+* ``--version`` - should always apply
+* ``--help`` -- should always apply
```
Comment (mr-c) [inline doc/dev/scripts-and-sandbox.txt]: one dash or two before the 'should'; be consistent please :-)

**Comment by mr-c** [comment]
Otherwise this LGTM

**Comment by ctb** [comment]
Ready for review (& merge?)

**Pull request merged**